how-to/add_controller.rst

```diff
@@ -1,119 +1,120 @@
+    CONTROLLER_NAME="workload-lxd"
+    CLOUDINIT_TEMPLATE=$'cloudinit-userdata: |
+    preruncmd:
+        - echo "%s    test-jimm.localhost" >> /etc/hosts
+    ca-certs:
+        trusted:
+        - |\n%s'
+    printf "$CLOUDINIT_TEMPLATE" "$(lxc network get lxdbr0 ipv4.address | cut -f1 -d/)" "$(cat /usr/local/share/ca-certificates/jimm-test.crt | sed -e 's/^/\t  /')" > "${CLOUDINIT_FILE}"
+    juju bootstrap lxd "${CONTROLLER_NAME}" --config "${CLOUDINIT_FILE}" --config login-token-refresh-url=https://test-jimm.localhost/.well-known/jwks.json --debug 
 
-    ``juju deploy  ./bundle.yaml --overlay ./overlay-certbot.yaml --map-machines=existing``
+The set of commands will do the following:
 
-9. Once the bundle has been deployed, get the public IP of the ``haproxy/0`` unit: 
+- Create a cloud-init template, cloud-init provisions the LXD container that Juju will use.
+- The cloud-init script will create an entry in ``/etc/hosts`` to point ``test-jimm.localhost`` to the LXD bridge address in order to route this request to your host network.
+- The cloud-init script will add the CA cert in ``/usr/local/share/ca-certificates/jimm-test.crt`` to the machine. If you've placed JIMM's CA cert elsewhere, please update this file location.
+- Finally the bash script will bootstrap Juju and configure it to communicate with JIMM.
 
-    ``juju status  --format json | jq '.applications.haproxy.units["haproxy/0"]["public-address"]'``
+Next, we will create a network relay to forward traffic from our host network through to the Juju server running in a LXC container.
 
-10.  Go to the `Route 53 dashboard <https://us-east-1.console.aws.amazon.com/route53/v2/home>`__.
+.. note::
+    The network relay relies on the ``socat`` application running continuously in the background.  
+    The application will need to be run again between system reboots.
 
-11.  Add an A record for the deployed controller and the DNS name specified in step 1 with the IP obtained in step 9.
+.. code:: bash
 
-12.  Obtain a valid certificate for the deployed candid by running: 
+    JUJU_ADDRESS=$(juju show-controller workload-lxd --format yaml | yq .workload-lxd.details.api-endpoints.[0])
+    socat tcp-listen:8001,reuseaddr,fork tcp:$JUJU_ADDRESS
 
-    ``juju run-action --wait certbot/0 get-certificate  agree-tos=true aws-access-key-id=<Access key ID> aws-secret-access-key=<Secret access key> domains=<dns name specified in step 1 (jimm.canonical.example.com)> email=<Your email address>  plugin=dns-route53``
+To test the relay is working run the following command which should return a HTTP 400 response code.
 
-13.  Install the JAAS snap that you download from `here <https://drive.google.com/file/d/1LiOvVpVQ13V3x3l2PhgS2fTHDUtCEe7p/view?usp=sharing>`_.
+.. code:: bash
 
-14. To add the bootstrapped controller to JIMM we need to create a controller-information document. To do this, run the following command:
+    curl https://localhost:8001 -k -I
```

Comment:
`-I` sets the request method to `HEAD`. I think it's better to use `-i` (to just do a `GET` and print the headers). Also, like this would be nicer:

```
curl -ki https://localhost:8001
```